clippy_lints/src/loops.rs

```diff
@@ -800,29 +800,126 @@ enum OffsetSign {
 }
 
 struct Offset {
-    value: String,
+    value: MinifyingSugg<'static>,
     sign: OffsetSign,
 }
 
 impl Offset {
-    fn negative(value: String) -> Self {
+    fn negative(value: MinifyingSugg<'static>) -> Self {
         Self {
             value,
             sign: OffsetSign::Negative,
         }
     }
 
-    fn positive(value: String) -> Self {
+    fn positive(value: MinifyingSugg<'static>) -> Self {
         Self {
             value,
             sign: OffsetSign::Positive,
         }
     }
+
+    fn empty() -> Self {
+        Self::positive(MinifyingSugg::non_paren("0"))
+    }
+}
+
+fn apply_offset(lhs: &MinifyingSugg<'static>, rhs: &Offset) -> MinifyingSugg<'static> {
+    match rhs.sign {
+        OffsetSign::Positive => lhs + &rhs.value,
+        OffsetSign::Negative => lhs - &rhs.value,
+    }
+}
+
+#[derive(Clone)]
+struct MinifyingSugg<'a>(sugg::Sugg<'a>);
```

Comment:
I'm not sure if I agree. Firstly, for `self.0`, most of `self.0` (and `rhs.0`) are where `self` is a reference so it would become `&**self`; and, two `rhs.0.clone()` would be `(**rhs).clone()` due to their precedence. Secondly, there're just few places where auto-dereferencing could happen; probably, only when I made `MinifyingSugg::as_str` the method of `Sugg`.

Reply:
Alright, I'm convinced 👍